private/util/content_md5.go

```diff
@@ -0,0 +1,78 @@
+package util
+
+import (
+	"crypto/md5"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/internal/sdkio"
+)
+
+const contentMD5Header = "Content-Md5"
+
+// ContentMD5 computes and sets the HTTP Content-MD5 header for requests that
+// require it.
+func ContentMD5(r *request.Request) {
```

Comment:
Might be better to clarify the purpose of this function by changing its name to, `AddBodyContentMD5Handler`

it would also be good to have a unit test for this behavior.